packages/react/src/Footnotes/Footnotes.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import {ReplyIcon} from '@primer/octicons-react'
+import clsx from 'clsx'
+
+import {Heading, InlineLink, Text} from '../'
+import type {BaseProps} from '../component-helpers'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/footnotes/colors-with-modes.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './Footnotes.module.css'
+
+export const FootnotesTags = ['div', 'ol'] as const
+
+type FootnotesRestrictedPolymorphism =
+  | (React.HTMLAttributes<HTMLDivElement> & BaseProps<HTMLDivElement> & {as?: 'div'})
+  | (React.HTMLAttributes<HTMLParagraphElement> & BaseProps<HTMLOListElement> & {as?: 'ol'})
+
+type FootnotesProps = {
+  as?: (typeof FootnotesTags)[number]
+  visuallyHiddenHeading?: string
+} & FootnotesRestrictedPolymorphism
+
+function Root({as = 'ol', className, children, visuallyHiddenHeading = 'Footnotes', ...rest}: FootnotesProps) {
+  if (as === 'div') {
+    const filteredChildren = React.Children.toArray(children).map(child => {
+      if (React.isValidElement(child) && typeof child.type !== 'string') {
+        if (child.type === Item) {
+          return React.cloneElement(child, {
+            variant: 'disclaimer',
+            ...child.props,
+          })
+        }
+      }
+    })
+    return (
+      <>
+        <Heading className="visually-hidden" as="h2" id="footnote-label">
+          {visuallyHiddenHeading}
+        </Heading>
+        <div
+          className={clsx(styles.Footnotes, styles['Footnotes--variant-disclaimer'], className)}
+          {...(rest as BaseProps<HTMLDivElement>)}
+        >
+          {filteredChildren}
+        </div>
+      </>
+    )
+  }
+
+  return (
+    <>
+      <Heading className="visually-hidden" as="h2" id="footnote-label">
+        {visuallyHiddenHeading}
+      </Heading>
+      <ol
+        className={clsx(styles.Footnotes, styles['Footnotes--variant-citations'], className)}
+        {...(rest as BaseProps<HTMLOListElement>)}
+      >
+        {children}
```

Comment:
Do you think we need to guard against someone passing `as="ol"` with `variant="disclaimer"`? That would render an `<ol>` with `<Text>` children.

Reply:
Good question. variant isn't documented anywhere outside of source code, and I'm using it as an internal-only prop. Bit of a trade-off between adding more code to guard it vs the managing the complexity of adding them.

In this case, because its undocumented I think i prefer to not add the code, though I have now prepended `_` to the start to make it clearer for TS users. Maybe this is enough of a guard? Could also go more heavy handed, add warnings, pass the `as` prop down and add conditional logic to prevent this configuration from being possible but we could always add that down the line too if we're seeing a lot of misuse.